src/expr/expr_node.rs

```diff
@@ -36,6 +39,10 @@ macro_rules! unary_fn {
     };
 }
 
+pub fn log4_roundup(n: usize) -> usize {
+    std::ops::Div::div((usize::BITS - n.leading_zeros()) as f64, 2.0).ceil() as usize
```

Comment:
Can this be simplified to `((n - 1).ilog2() + 1).ceil_div(2)`? If this works, then I would slightly prefer the name `log4_ceil` for the function.

Reply:
Oh this needs a branch for the edge case `n <= 1`.